x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/planner/QueryTranslatorTests.java

```diff
@@ -152,4 +163,37 @@ public void testRanges() {
             .*must.*esql_single_value":\\{"field":"unsigned_long\"""" + """
             .*"range":\\{"unsigned_long":\\{"gte":2147483648,.*"range":\\{"unsigned_long":\\{"lte":2147483650,.*"""));
     }
+
+    public void testIPs() {
+        // Nothing to combine
+        assertQueryTranslationIPs("""
+            FROM hosts | WHERE CIDR_MATCH(ip0, "127.0.0.1") OR CIDR_MATCH(ip0, "127.0.0.3") AND CIDR_MATCH(ip1, "fe80::cae2:65ff:fece:fec0")
+            """, matchesRegex("""
+            .*bool.*should.*""" + """
+            esql_single_value":\\{"field":"ip0".*"terms":\\{"ip0":\\["127.0.0.1".*""" + """
+            .*bool.*must.*""" + """
+            esql_single_value":\\{"field":"ip0".*"terms":\\{"ip0":\\["127.0.0.3".*""" + """
+            esql_single_value":\\{"field":"ip1".*"terms":\\{"ip1":\\["fe80::cae2:65ff:fece:fec0".*"""));
+
+        // Combine Equals, In and CIDRMatch on IP type
+        assertQueryTranslationIPs("""
+            FROM hosts | WHERE host == "alpha" OR host == "gamma" OR CIDR_MATCH(ip1, "127.0.0.2/32") OR CIDR_MATCH(ip1, "127.0.0.3/32") \
+            OR card IN ("eth0", "eth1") OR card == "lo0" OR CIDR_MATCH(ip0, "127.0.0.1") OR \
+            CIDR_MATCH(ip0, "fe80::cae2:65ff:fece:feb9") OR host == "beta\"""", matchesRegex("""
+            .*bool.*should.*""" + """
+            esql_single_value":\\{"field":"host".*"terms":\\{"host":\\["alpha","gamma","beta".*""" + """
+            esql_single_value":\\{"field":"card".*"terms":\\{"card":\\["eth0","eth1","lo0".*""" + """
+            esql_single_value":\\{"field":"ip1".*"terms":\\{"ip1":\\["127.0.0.2/32","127.0.0.3/32".*""" + """
+            esql_single_value":\\{"field":"ip0".*"terms":\\{"ip0":\\["127.0.0.1","fe80::cae2:65ff:fece:feb9".*"""));
+
+        assertQueryTranslationIPs("""
+            FROM hosts | WHERE host == "alpha" OR host == "gamma" OR ip1 IN ("127.0.0.2"::ip) OR CIDR_MATCH(ip1, "127.0.0.3/32") \
```

Comment:
This is interesting! I didn't catch this subtlety when I created that query in my previous review. `ip1 IN ("127.0.0.2/32"::ip)` generates a warning because `127.0.0.2/32` is not a valid `ip`, BUT `terms` query from ES accepts a string like `127.0.0.2/32`. Which means that, with ES|QL, if one wants to do `ip1 IN ("127.0.0.2/32"::ip)` (which gets translated into a `terms` query to be pushed down to Lucene) should actually use `cidr_match("127.0.0.2/32"::ip)` (that gets translated in the same `terms` query as before).

And this is ok!